In [2]:
from sklearn import datasets

In [8]:
X_digits, Y_digits = datasets.load_digits(return_X_y=True)
X_breast_cancer, Y_breast_cancer = datasets.load_breast_cancer(return_X_y=True)

In [9]:
X_digits

array([[ 0.,  0.,  5., ...,  0.,  0.,  0.],
       [ 0.,  0.,  0., ..., 10.,  0.,  0.],
       [ 0.,  0.,  0., ..., 16.,  9.,  0.],
       ...,
       [ 0.,  0.,  1., ...,  6.,  0.,  0.],
       [ 0.,  0.,  2., ..., 12.,  0.,  0.],
       [ 0.,  0., 10., ..., 12.,  1.,  0.]])

In [10]:
X_breast_cancer

array([[1.799e+01, 1.038e+01, 1.228e+02, ..., 2.654e-01, 4.601e-01,
        1.189e-01],
       [2.057e+01, 1.777e+01, 1.329e+02, ..., 1.860e-01, 2.750e-01,
        8.902e-02],
       [1.969e+01, 2.125e+01, 1.300e+02, ..., 2.430e-01, 3.613e-01,
        8.758e-02],
       ...,
       [1.660e+01, 2.808e+01, 1.083e+02, ..., 1.418e-01, 2.218e-01,
        7.820e-02],
       [2.060e+01, 2.933e+01, 1.401e+02, ..., 2.650e-01, 4.087e-01,
        1.240e-01],
       [7.760e+00, 2.454e+01, 4.792e+01, ..., 0.000e+00, 2.871e-01,
        7.039e-02]])

In [17]:
from sklearn import cross_validation
from sklearn import naive_bayes

In [18]:
MultinomialNB = naive_bayes.MultinomialNB()
GaussianNB = naive_bayes.GaussianNB()
BernoulliNB = naive_bayes.BernoulliNB()

In [19]:
dig_mult = cross_validation.cross_val_score(MultinomialNB, X_digits, y = Y_digits)
dig_gauss = cross_validation.cross_val_score(GaussianNB, X_digits, y = Y_digits)
dig_bernoulli = cross_validation.cross_val_score(BernoulliNB, X_digits, y = Y_digits)

In [20]:
cancer_mult = cross_validation.cross_val_score(MultinomialNB, X_breast_cancer, y = Y_breast_cancer)
cancer_gauss = cross_validation.cross_val_score(GaussianNB, X_breast_cancer, y = Y_breast_cancer)
cancer_bernoulli = cross_validation.cross_val_score(BernoulliNB, X_breast_cancer, y = Y_breast_cancer)

In [23]:
arr_val = [dig_mult.mean(), dig_gauss.mean(), dig_bernoulli.mean(), 
 cancer_mult.mean(), cancer_gauss.mean(), cancer_bernoulli.mean()]
arr_val

[0.8708771489735053,
 0.8186003803550138,
 0.8258236507780582,
 0.8945790401930752,
 0.9367492806089297,
 0.6274204028589994]

In [26]:
max(arr_val[4:])

0.9367492806089297

In [28]:
max(arr_val[:3])

0.8708771489735053

Вопрос 1

Каким получилось максимальное качество классификации на датасете breast_cancer?

0.9367492806089297

Вопрос 2

Каким получилось максимальное качество классификации на датасете digits?

0.8708771489735053

Вопрос 3

Выберите верные утверждения и запишите их номера через пробел (в порядке возрастания номера):

1) На вещественных признаках лучше всего сработал наивный байесовский классификатор с распределением Бернулли

2) На вещественных признаках лучше всего сработал наивный байесовский классификатор с мультиномиальным распределением

3) Мультиномиальное распределение лучше показало себя на выборке с целыми неотрицательными значениями признаков

4) На вещественных признаках лучше всего сработало нормальное распределение

4